iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -325,7 +339,15 @@ private void commitTable(FileIO io, ExecutorService executor, JobContext jobCont
     LOG.info("Committing job has started for table: {}, using location: {}",
         table, generateJobLocation(location, conf, jobContext.getJobID()));
 
-    int numTasks = SessionStateUtil.getCommitInfo(conf, name).map(info -> info.getTaskNum()).orElseGet(() -> {
+    Optional<SessionStateUtil.CommitInfo> commitInfo;
+    if (SessionStateUtil.getCommitInfo(conf, name).isPresent()) {
+      commitInfo = SessionStateUtil.getCommitInfo(conf, name).get()
+              .stream().filter(ci -> ci.getJobIdStr().equals(jobContext.getJobID().toString())).findFirst();
```

Comment:
Are we sure that the first `commitInfo` has the same number of tasks as all of them?

Reply:
AFAIK only one `CommitInfo` object should be associated to a jobContext.

Reply:
Then why `.findFirst()`?

Reply:
Replaced the `List<CommitInfo>` to `Map<String, CommitInfo>` where the key is the string representation of the `jobId` associated to the `CommitInfo`.